### DISCUSSION TITLE
Can we build using boringssl?

### DISCUSSION THREAD

**Discussion by manuelbuil** [strongswan/strongswan#1270] [Q&A]
When looking at https://docs.strongswan.org/docs/5.9/plugins/plugins.html, I don't see boringssl as an available crypto backend. However, when looking at the code, I can see the variable OPENSSL_IS_BORINGSSL: https://github.com/strongswan/strongswan/blob/df18934d2075029c2b7e2fb5a2821de836eb71bb/src/libstrongswan/plugins/openssl/openssl_rsa_public_key.c#L129 which confuses me. I was not able to find any doc explaining how to activate that variable or describing how to build using boringssl instead of openssl. Therefore, the question: can we make strongswan use boringssl? Thanks!

**Comment by Thermi** [comment]
Yes.
I think it's built that way for Android. See https://github.com/strongswan/strongswan/tree/master/src/frontends/android

**Reply by strongX509** [reply]
As Thermi already wrote, we're using boringssl for our Android NDK build but it should also work in a normal Linux environment when the boringssl library and the matching header files are in place.
